packages/yarnpkg-pnp/sources/loader/makeApi.ts

```diff
@@ -856,18 +868,22 @@ export function makeApi(runtimeState: RuntimeState, opts: MakeApiOptions): PnpAp
         ? isPathIgnored(issuer)
         : false;
 
+    const wrapError = <T>(fn: () => T, errorCode: ErrorCode) => {
+      try {
+        return fn();
+      } catch (resolutionError) {
+        if (resolutionError.pnpCode === errorCode)
+          Object.assign(resolutionError.data, {request: getPathForDisplay(request), issuer: issuer && getPathForDisplay(issuer)});
+
+        throw resolutionError;
+      }
+    };
+
     const remappedPath = (!considerBuiltins || !isBuiltinModule(request)) && !isIssuerIgnored()
-      ? resolveUnqualifiedExport(request, unqualifiedPath, conditions)
+      ? wrapError(() => resolveUnqualifiedExport(request, unqualifiedPath, conditions), ErrorCode.EXPORTS_RESOLUTION_FAILED)
       : unqualifiedPath;
 
-    try {
-      return resolveUnqualified(remappedPath, {extensions});
-    } catch (resolutionError) {
-      if (resolutionError.pnpCode === `QUALIFIED_PATH_RESOLUTION_FAILED`)
-        Object.assign(resolutionError.data, {request: getPathForDisplay(request), issuer: issuer && getPathForDisplay(issuer)});
-
-      throw resolutionError;
-    }
+    return wrapError(() => resolveUnqualified(remappedPath, {extensions}), ErrorCode.QUALIFIED_PATH_RESOLUTION_FAILED);
```

Comment:
Perhaps this double-wrap is unnecessary and we could just try/catch those five lines in a single block (and replace the check on specific pnpCode to cover all PnP exceptions, regardless of their type)?

Reply:
Done :+1: